analysis/semsim/unifiedmodel_v2.py

```diff
@@ -0,0 +1,401 @@
+                encoded_text_type[f'{text_type}_{encoded_key}'] = encoded_text_type.pop(encoded_key) if not self.streaming_dataset else encoded_text_type.pop(
+                    encoded_key).squeeze(0)  # e.g. input_ids -> title_input_ids so we have separate input_ids for each text_type
+            del batch[f'regret_{text_type}']
+            del batch[f'recommendation_{text_type}']
+            batch.update(encoded_text_type)
+
+        # copy needed columns from original dataset to the encoded dataset
+        if self.use_scalar_features:
+            for scalar_feat in self.scalar_features:
+                batch[scalar_feat] = torch.as_tensor(
+                    batch[scalar_feat]) if not self.streaming_dataset else torch.as_tensor(batch[scalar_feat]).squeeze(0)
+        if self.use_channel_embeddings:
+            for ch_emb in self.channel_embeddings:
+                batch[ch_emb] = torch.as_tensor(batch[ch_emb])
+        if self.label_col:
+            batch[self.label_col] = torch.as_tensor(
+                batch[self.label_col]) if not self.streaming_dataset else torch.as_tensor(batch[self.label_col]).squeeze(0)
+        return batch
+
+
+class RRUMV2(pl.LightningModule):
+    def __init__(self, text_types, scalar_features, label_col, optimizer_config, model_name_or_path, channel_embeddings=[], channel_embedding_dim=None, freeze_policy=None, pos_weight=None):
+        super().__init__()
+        self.save_hyperparameters()
+        self.text_types = text_types
+        self.scalar_features = scalar_features
+        self.label_col = label_col
+        self.optimizer_config = optimizer_config
+        self.model_name_or_path = model_name_or_path
+        self.channel_embeddings = channel_embeddings
+        self.transformer_models = nn.ModuleDict({}) if text_types else None
+        for t in self.text_types:
+            self.transformer_models[t] = Transformer(self.model_name_or_path)
+        if freeze_policy is not None:
+            for model in self.transformer_models.values():
+                for name, param in model.named_parameters():
+                    if freeze_policy(name):
+                        param.requires_grad = False
+        if self.transformer_models:
+            word_emb_dim = self.transformer_models[self.text_types[0]
+                                                   ].get_word_embedding_dimension()
+            self.pooler = Pooling(word_emb_dim, 'mean')
+            pooled_output_dim = self.pooler.get_sentence_embedding_dimension()
+
+        if self.channel_embeddings and channel_embedding_dim:
+            self.channel_emb_fc = nn.Linear(
+                channel_embedding_dim * len(self.channel_embeddings), channel_embedding_dim)
+
+        mlp_input_dim = (len(self.transformer_models) * pooled_output_dim if self.transformer_models else 0) + len(
+            self.scalar_features) + (channel_embedding_dim if channel_embedding_dim else 0)
+        self.fc1 = nn.Linear(mlp_input_dim, 1024)
```

Comment:
Is it a better practice to define the layer sizes as constants and refer to those in these lines?  I don't feel strongly though.